### ISSUE TITLE
bug: no-construct-in-interface is not working with generic types

### ISSUE THREAD

**Issue by ren-yamanashi** [ren-yamanashi/eslint-cdk-plugin#220]
### Description
The `no-construct-in-interface` ESLint rule is not detecting construct types when they are wrapped in generic types such as `Readonly<T>`, `Array<T>`, `Partial<T>`, etc.

### Example Code
```typescript
import { ManagedComputeEnvironmentBase } from "aws-cdk-lib/aws-batch";
import { Bucket } from "aws-cdk-lib/aws-s3";

interface MyConstructProps {
  readonly envBase: Readonly<ManagedComputeEnvironmentBase>; // not raising an error
  readonly buckets: Array<Bucket>;                           // not raising an error
  readonly partialEnv: Partial<ManagedComputeEnvironmentBase>; // not raising an error
}
```

### Expected Behavior
The rule should detect and report an error when CDK Construct types are used within generic type wrappers, similar to how it works for non-wrapped types.

### Actual Behavior
No error is reported when the construct type is wrapped in generic types.

### Environment
- eslint-cdk-plugin Version: 3.4.0
- Node.js Version: 22.15.0
- Language: TypeScript

### Related Issues
- #216 - Similar issue but focused on array syntax `Bucket[]` (fixed in #217)
- This issue is about generic type syntax like `Array<T>`, `Readonly<T>`, `Partial<T>`, etc.

### Additional Context
Note that #217 fixed the array bracket syntax `Bucket[]`, but generic type wrappers still don't trigger the rule. The rule should ideally unwrap generic types to check the underlying type arguments.
